src/IISIntegration/src/Microsoft.AspNetCore.Server.IISIntegration/IISSetupFilter.cs

```diff
@@ -28,8 +32,23 @@ public Action<IApplicationBuilder> Configure(Action<IApplicationBuilder> next)
                 app.UsePathBase(_pathBase);
                 app.UseForwardedHeaders();
                 app.UseMiddleware<IISMiddleware>(_pairingToken, _isWebsocketsSupported);
+                if (_serverAddresses != null)
+                {
+                    app.ServerFeatures.Set<IServerAddressesFeature>(new ServerAddressesFeature(_serverAddresses));
```

Comment:
Out of proc can't set IServerAddressesFeature, it will confuse Kestrel. What would be interesting for out of proc is setting the HTTPS_PORT.

Reply:
Seems to work fine if you set it in a startup filter. Who consumes HTTPS_PORT?

Reply:
No no no remove this.

HTTPS_PORT is consumed by the HTTPS redirect middleware.
https://github.com/aspnet/AspNetCore/blob/c95ee2b051814b787b07f55ff224d03d550aafeb/src/Middleware/HttpsPolicy/src/HttpsRedirectionMiddleware.cs#L111

Reply:
Gotcha